### PR TITLE
4 client refactor to ts

### LINKED ISSUE #4: [client] Refactor to TS

### PR DESCRIPTION
**Pull request by khomch**

### CODE REVIEW

File: client/src/apiService.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+  } catch (error) {
+    console.log('Issue occured on logout: ' + error);
+  }
+}
+
+async function addCategory(category: any) {
```
Comment (simplyjuanc) [inline client/src/apiService.tsx:65]: I see some `any`s on this file XD

File: client/src/components/home.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      return;
+    } else {
+      const updatedUser = await addCategory({
+        category: categoryInput.toLowerCase(),
+      });
+      setLoggedUser(updatedUser);
+      setAddCategoryClicked(false);
+      setCategoryInput('');
+    }
+  }
```
Comment (simplyjuanc) [inline client/src/components/home.tsx:64]: Could be made into a single handler function, but open question though: How much optimisation do we want to include in the code?

File: client/src/components/login.tsx [hunk 3]
```diff
@@ -1,18 +1,17 @@
...
-  const setLogged = useCombinedStore(state => state.fetchLoggedUser);
+  const setLogged = useCombinedStore((state) => state.fetchLoggedUser);
 
   const navigate = useNavigate();
 
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
 
-  async function handleSubmit(event) {
+  async function handleSubmit(event: FormEvent) {
```
Comment (simplyjuanc) [inline client/src/components/login.tsx:14]: Same as above

**Review by simplyjuanc — APPROVED** [review]

File: client/src/components/home.tsx (hunk 2 above)
Comment (khomch) [reply client/src/components/home.tsx:64]: I left all func/components as they are to split logic as separate issue & pull request.

File: client/src/apiService.tsx (hunk 1 above)
Comment (khomch) [reply client/src/apiService.tsx:65]: Good catch!

**Pull request merged**